### ISSUE TITLE
"IPVS: Protocol not available" running 1.3.0 or 1.3.2 on CentOS 7.3.1611

### ISSUE THREAD

**Issue by Bernd2** [acassen/keepalived#489]
Running keepalived 1.3.0 or 1.3.2 (latter built from git) throws following error message when being run on CentOS 7.3.1611 (3.10.0-514.2.2.el7.x86_64):

IPVS: Protocol not available

**Comment by kustodian** [comment]
I'm having the exact same problem on the same CentOS version, but with version 1.2.24. It's like keepalived doesn't load `ip_vs` module on time, but the problem is that after a few seconds it does load it, but then the healthcheckers stop working and start reporting `IPVS: Service not defined`.

Here is the output in the syslog:
```
Jan  5 19:33:49 lb-02 Keepalived[4374]: Starting Keepalived v1.2.24 (11/30,2016)
Jan  5 19:33:49 lb-02 Keepalived[4374]: Opening file '/etc/keepalived/keepalived.conf'.
Jan  5 19:33:49 lb-02 Keepalived[4375]: Starting Healthcheck child process, pid=4376
Jan  5 19:33:49 lb-02 Keepalived[4375]: Starting VRRP child process, pid=4377
Jan  5 19:33:49 lb-02 systemd: Started LVS and VRRP High Availability Monitor.
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]: Netlink reflector reports IP 10.10.100.43 added
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]: Registering Kernel netlink reflector
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]: Registering Kernel netlink command channel
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]: Registering gratuitous ARP shared channel
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]: Opening file '/etc/keepalived/keepalived.conf'.
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]: (VI_1): Specifying lvs_sync_daemon_interface against a vrrp is deprecated.
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]:        Please use global lvs_sync_daemon
Jan  5 19:33:49 lb-02 Keepalived_healthcheckers[4376]: Netlink reflector reports IP 10.10.100.43 added
Jan  5 19:33:49 lb-02 Keepalived_healthcheckers[4376]: Registering Kernel netlink reflector
Jan  5 19:33:49 lb-02 Keepalived_healthcheckers[4376]: Registering Kernel netlink command channel
Jan  5 19:33:49 lb-02 Keepalived_healthcheckers[4376]: Opening file '/etc/keepalived/keepalived.conf'.
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]: VRRP_Instance(VI_1) removing protocol VIPs.
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]: Using LinkWatch kernel netlink reflector...
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]: IPVS: Protocol not available
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]: VRRP_Instance(VI_1) Entering BACKUP STATE
Jan  5 19:33:49 lb-02 Keepalived_vrrp[4377]: VRRP sockpool: [ifindex(2), proto(112), unicast(1), fd(11,12)]
Jan  5 19:33:49 lb-02 Keepalived_healthcheckers[4376]: IPVS: Protocol not available
Jan  5 19:33:49 lb-02 Keepalived_healthcheckers[4376]: IPVS: Protocol not available
Jan  5 19:33:49 lb-02 Keepalived_healthcheckers[4376]: Using LinkWatch kernel netlink reflector...
Jan  5 19:33:49 lb-02 Keepalived_healthcheckers[4376]: Activating healthchecker for service [10.10.100.50]:1111
Jan  5 19:33:49 lb-02 Keepalived_healthcheckers[4376]: Activating healthchecker for service [10.10.100.51]:1111
Jan  5 19:33:49 lb-02 Keepalived_healthcheckers[4376]: Activating healthchecker for service [10.10.100.50]:2222
Jan  5 19:33:49 lb-02 Keepalived_healthcheckers[4376]: Activating healthchecker for service [10.10.100.51]:2222
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: TCP connection to [10.10.100.50]:1111 success.
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: Adding service [10.10.100.50]:1111 to VS [10.10.100.44]:1111
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: IPVS: Protocol not available
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: TCP connection to [10.10.100.51]:1111 success.
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: Adding service [10.10.100.51]:1111 to VS [10.10.100.44]:1111
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: IPVS: Protocol not available
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: HTTP status code success to [10.10.100.50]:2222 url(1).
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: Remote Web server [10.10.100.50]:2222 succeed on service.
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: Adding service [10.10.100.50]:2222 to VS [10.10.100.44]:2222
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: IPVS: Protocol not available
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: HTTP status code success to [10.10.100.51]:2222 url(1).
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: Remote Web server [10.10.100.51]:2222 succeed on service.
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: Adding service [10.10.100.51]:2222 to VS [10.10.100.44]:2222
Jan  5 19:33:50 lb-02 Keepalived_healthcheckers[4376]: IPVS: Protocol not available
Jan  5 19:33:53 lb-02 Keepalived_healthcheckers[4376]: TCP connection to [10.10.100.50]:1111 success.
Jan  5 19:33:53 lb-02 Keepalived_healthcheckers[4376]: Adding service [10.10.100.50]:1111 to VS [10.10.100.44]:1111
Jan  5 19:33:53 lb-02 Keepalived_healthcheckers[4376]: IPVS: Protocol not available
Jan  5 19:33:53 lb-02 Keepalived_healthcheckers[4376]: TCP connection to [10.10.100.51]:1111 success.
Jan  5 19:33:53 lb-02 Keepalived_healthcheckers[4376]: Adding service [10.10.100.51]:1111 to VS [10.10.100.44]:1111
...
...
...
Jan  5 19:35:22 lb-02 kernel: IPVS: Registered protocols (TCP, UDP, SCTP, AH, ESP)
Jan  5 19:35:22 lb-02 kernel: IPVS: Connection hash table configured (size=4096, memory=64Kbytes)
Jan  5 19:35:22 lb-02 kernel: IPVS: Creating netns size=2040 id=0
Jan  5 19:35:22 lb-02 kernel: IPVS: ipvs loaded.
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: TCP connection to [10.10.100.50]:1111 success.
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: Adding service [10.10.100.50]:1111 to VS [10.10.100.44]:1111
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: IPVS: Service not defined
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: TCP connection to [10.10.100.51]:1111 success.
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: Adding service [10.10.100.51]:1111 to VS [10.10.100.44]:1111
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: IPVS: Service not defined
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: HTTP status code success to [10.10.100.50]:2222 url(1).
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: Remote Web server [10.10.100.50]:2222 succeed on service.
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: Adding service [10.10.100.50]:2222 to VS [10.10.100.44]:2222
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: IPVS: Service not defined
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: HTTP status code success to [10.10.100.51]:2222 url(1).
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: Remote Web server [10.10.100.51]:2222 succeed on service.
Jan  5 19:35:23 lb-02 Keepalived_healthcheckers[4376]: Adding service [10.10.100.51]:2222 to VS [10.10.100.44]:2222
...
...
...
```
It's easy to reproduce this, either reboot the machine, or:
```bash
# Stop keepalived
systemct stop keepalived

# Unload ip_vs_rr and ip_vs modules
rmmod ip_vs_rr ip_vs

# Start keepalived
systemct start keepalived
```

This problem doesn't exist on CentOS 7.2, keepalived loads the module without any problems.

Current workaround is to create a file: `/etc/modules-load.d/ip_vs.conf` with the following contents
```
# Load ip_vs because keepalived doesn't load it correctly
ip_vs
```
So the module will be loaded on boot, or without rebooting just load the module manually `modprobe ip_vs`.

**Comment by pqarmitage** [comment]
This is resolved with commit 4438875. Apologies for the problem.
